cluster/environments/azure-simple/main.tf

```diff
@@ -1,61 +1,29 @@
-provider "azurerm" {
-  subscription_id = "${var.subscription_id}"
-  client_id       = "${var.service_principal_id}"
-  client_secret   = "${var.service_principal_secret}"
-  tenant_id       = "${var.tenant_id}"
+terraform {
+  backend "azurerm" {}
 }
 
-# terraform {
-#    backend "azurerm" {
-#    }
-# }
-
-data "azurerm_subscription" "current" {
-  subscription_id = "${var.subscription_id}"
-}
-
-module "vnet" {
-  source = "../../azure/vnet"
-
-  resource_group_name     = "${var.resource_group_name}"
-  resource_group_location = "${var.resource_group_location}"
-  subnet_names            = ["${var.cluster_name}-aks-subnet"]
-
-  tags = {
-    environment = "azure-simple"
-  }
+resource "azurerm_resource_group" "cluster_rg" {
+  name     = "${var.resource_group_name}"
+  location = "${var.resource_group_location}"
 }
 
-module "aks" {
-  source = "../../azure/aks"
+module "aks-gitops" {
+  source = "github.com/timfpark/bedrock//cluster/azure/aks-gitops?ref=aks_gitops"
 
-  resource_group_location  = "${var.resource_group_location}"
-  resource_group_name      = "${var.resource_group_name}"
-  cluster_name             = "${var.cluster_name}"
+  acr_enabled              = "${var.acr_enabled}"
   agent_vm_count           = "${var.agent_vm_count}"
   agent_vm_size            = "${var.agent_vm_size}"
+  cluster_name             = "${var.cluster_name}"
   dns_prefix               = "${var.dns_prefix}"
-  vnet_subnet_id           = "${module.vnet.vnet_subnet_ids[0]}"
-  ssh_public_key           = "${var.ssh_public_key}"
+  flux_recreate            = "${var.flux_recreate}"
+  gitops_ssh_url           = "${var.gitops_ssh_url}"
+  gitops_ssh_key           = "${var.gitops_ssh_key}"
+  gitops_path              = "${var.gitops_path}"
+  gitops_poll_interval     = "${var.gitops_poll_interval}"
+  resource_group_location  = "${var.resource_group_location}"
+  resource_group_name      = "${azurerm_resource_group.cluster_rg.name}"
   service_principal_id     = "${var.service_principal_id}"
   service_principal_secret = "${var.service_principal_secret}"
-  kubeconfig_recreate      = ""
-}
-
-module "flux" {
-  source = "../../common/flux"
-
-  gitops_ssh_url      = "${var.gitops_ssh_url}"
-  gitops_ssh_key      = "${var.gitops_ssh_key}"
-  flux_recreate       = "${var.flux_recreate}"
-  kubeconfig_complete = "${module.aks.kubeconfig_done}"
-  flux_clone_dir      = "${var.cluster_name}-flux"
-  gitops_path            = "${var.gitops_path}"
-}
-
-module "kubediff" {
-    source = "../../common/kubediff"
-
-    kubeconfig_complete       = "${module.aks.kubeconfig_done}"
-    gitops_ssh_url            = "${var.gitops_ssh_url}"
+  ssh_public_key           = "${var.ssh_public_key}"
+  vnet_subnet_id           = "${var.vnet_subnet_id}"
```

Comment:
This should be gathered from the provisioning, not a var

Reply:
Yes, I added a vnet provisioner and obtain the subnet id from it.